fix: address issues with eslint on CI

Following issues need to be addressed to promote smooth execution of eslint on CI:

- Upgrade Node version and avoid additional installation of eslint specific dependancies.
- Remove the `continue-on-true` flag as the repository is now stable and developers should address lint issues triggered for their PR.

**ERROR LOGS ON LINT JOB FROM GITHUB ACTIONS:**

```
Run npm run lint

> maidr-ts@[1](https://github.com/xability/maidr-ts/actions/runs/13574276125/job/37946578064#step:6:1).0.0 lint
> eslint .


Oops! Something went wrong! :(

ESLint: 8.[5](https://github.com/xability/maidr-ts/actions/runs/13574276125/job/37946578064#step:6:6)7.1

No files matching the pattern "." were found.
Please check for typing mistakes in the pattern.

Error: Process completed with exit code 2.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -22,4 +22,3 @@ jobs:
 
       - name: Run ESLint
         run: npm run lint
-        continue-on-error: true

